src/Application.vala

```diff
@@ -107,6 +112,11 @@ public class Akira.Application : Gtk.Application {
     }
 
     protected override void activate () {
+        // Interrupt if we have at least one existing window.
+        if (windows.length () > 0) {
+            return;
+        }
+
```

Comment:
This will not fix the double windows since it is being called before the windows are even created and this is not the only problem the user may have an akira window already opened so the windows will be more than 0 and that will not let the user open a new window 
so if you mean to do this then the code that is responsible of not allowing same file not opening and the code which is responsible to load the file in an empty window is non reachable

Reply:
good point

Reply:
It should be fixed now.